base/array.jl

```diff
@@ -1987,3 +1987,162 @@ julia> symdiff([1,2,3],[3,4,5],[4,5,6])
 ```
 """
 symdiff(a, b, rest...) = symdiff(a, symdiff(b, rest...))
+
+## replace/replace! ##
+
+# NOTE: to implement replace! and replace for a new type T, it's enough to define:
+# function _replace!(pred::Callable, new::Callable, A::T, n::Int)
+
+"""
+    replace!(A, old_new::Pair...; [n::Integer])
+
+For each pair `old=>new` in `old_new`, replace all occurrences
+of `old` in collection `A` by `new`.
+If `n` is specified, then replace at most `n` occurrences in total.
+See also [`replace`](@ref replace(A, old_new::Pair...)).
+
+# Examples
+```jldoctest
+julia> replace!([1, 2, 1, 3], 1=>0, 2=>4, n=2)
+4-element Array{Int64,1}:
+ 0
+ 4
+ 1
+ 3
+
+julia> replace!(Set([1, 2, 3]), 1=>0)
+Set([0, 2, 3])
+```
+"""
+replace!(A, old_new::Pair...; n::Integer=typemax(Int)) = _replace!(A, eltype(A), n, old_new...)
+
+_replace!(A, ::Type{K}, n::Integer, old_new::Pair...) where {K} = _replace!(A, n) do x
+    for on in old_new
+        first(on) == x && return Nullable{K}(last(on))
+    end
+    Nullable{K}()
+end
+
+"""
+    replace!(pred::Function, A, new; [n::Integer])
+
+Replace all occurrences `x` in collection `A` for which `pred(x)` is true
+by `new`.
+
+# Examples
+```jldoctest
+julia> A = [1, 2, 3, 1];
+
+julia> replace!(isodd, A, 0, n=2); a
```

Comment:
`a` does not exist. Just remove it (I think we discussed this somewhere already).